doc/readme-pvs-studio.md

```diff
@@ -0,0 +1,25 @@
+# How to integrate PVS-Studio
+
+We obtained a free license of PVS-Studio due to the open source nature of Exiv2.
+
+## CMake integration
+
+To check a project configured with CMake, such as Exiv2, we need to generate the JSON compilation database. 
+
+```bash
+# Under an already configured "buildXXX" directory
+cd buildXXX
+cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ...
```

Comment:
Should we just make this a default CMake option? 
It doesn't hurt and most editors/IDE somehow go looking for such file anyway

Reply:
Humm ... I'm not sure about this. I'm pretty sure that it does not add much "extra" time to configure the project, but I would not add more things to the default CMake configuration, unless it is benefitial for all the users/consumers/developers.

In my experience, when you start adding many of these "little nice things" to your CMake code,  you decrease gradually the speed of the configuration phase in your project. At some point in the future (maybe after years) you might start to wonder why the configuration step in your CMake project is so slow (specially on Windows!).

However, I am open to include this change if you and others can really benefit from this. Could you ellaborate a bit more in which scenarios, editors, IDEs you think this might be useful?

Reply:
I usually use the `compile_commands.json` because my editor in the back uses the `clangd` language server for auto-completion and other nice things.           
I'm not too familiar with Visual Studo Code but they also have a C++ extension which is [very easy to set up with a that file](https://code.visualstudio.com/docs/cpp/faq-cpp#_how-do-i-get-intellisense-to-work-correctly).  

 I think for a lot C++ tooling this file has kinda become a standard input.

Reply:
I added that option by default after checking that it only adds few milliseconds more to the configuration step 😉 .